library/ziti.c

```diff
@@ -1783,3 +1796,93 @@ void ziti_queue_work(ziti_context ztx, ztx_work_f w, void *data) {
 
     uv_async_send(&ztx->w_async);
 }
+
+int ziti_context_init(ziti_context *ztx, const ziti_config *config) {
+    if (config == NULL || config->controller_url == NULL) {
+        ZITI_LOG(ERROR, "config or controller/tls has to be set");
+        return ZITI_INVALID_CONFIG;
+    }
+
+    ziti_context ctx = calloc(1, sizeof(*ctx));
+
+    if (strncmp(config->id.ca, "file://", strlen("file://")) == 0) {
+        struct tlsuv_url_s url;
+        if (tlsuv_parse_url(&url, config->id.ca) != 0) {
+            ZITI_LOG(ERROR, "invalid CA bundle reference[]");
+            return ZITI_INVALID_CONFIG;
+        }
+
+        char *ca = NULL;
+        size_t ca_len;
+        int rc = load_file(url.path, url.path_len, &ca, &ca_len);
+        if (rc == 0) {
+            FREE(ctx->config.id.ca);
```

Comment:
this seems suspect. the caller may be responsible for this pointer?

Reply:
all passed structs are `const` and their content is copied into `ziti_context`. this avoids ownership ambiguity